dask_cuda/proxify_host_file.py

```diff
@@ -123,31 +152,45 @@ class ProxiesOnDevice(Proxies):
     handle that multiple proxy objects can refer to the same underlying
     device memory object. Thus, we have to track aliasing and make sure
     we don't count down the memory usage prematurely.
+
+    Notice, we only track direct aliasing thus multiple proxy objects can
+    point to different non-overlapping parts of the same device buffer.
+    In this case the tally of the total device memory usage is incorrect.
     """
 
     def __init__(self):
         super().__init__()
-        self.proxy_id_to_dev_mems: Dict[int, Set[Hashable]] = {}
-        self.dev_mem_to_proxy_ids: DefaultDict[Hashable, Set[int]] = defaultdict(set)
+        self.proxy_id_to_dev_mems: Dict[int, Set[DeviceMemoryId]] = {}
+        self.dev_mem_to_proxy_ids: DefaultDict[DeviceMemoryId, Set[int]] = defaultdict(
+            set
+        )
 
-    def mem_usage_add(self, proxy: ProxyObject):
+    def mem_usage_add(self, proxy: ProxyObject) -> None:
         proxy_id = id(proxy)
         assert proxy_id not in self.proxy_id_to_dev_mems
         self.proxy_id_to_dev_mems[proxy_id] = set()
-        for dev_mem in proxy._pxy_get_device_memory_objects():
+        for dev_mem in get_device_memory_ids(proxy._pxy_get().obj):
             self.proxy_id_to_dev_mems[proxy_id].add(dev_mem)
             ps = self.dev_mem_to_proxy_ids[dev_mem]
             if len(ps) == 0:
-                self._mem_usage += sizeof(dev_mem)
+                self._mem_usage += dev_mem.nbytes
```

Comment:
Why is `sizeof` not suitable here anymore?

Reply:
This is just for performance, now that we know that `dev_mem` is a `class DeviceMemoryId` it is faster to query the size directly instead of implementing `DeviceMemoryId.__sizeof__()`.